ci-bench/src/main.rs

```diff
@@ -150,31 +157,101 @@ fn main() -> anyhow::Result<()> {
+                                config: ClientSideStepper::make_config(params, resumption_kind),
+                            },
+                            bench.kind,
+                        )
+                        .await
+                    }
+                }
+            })
+            .with_context(|| format!("{} crashed for {} side", bench.name(), side.as_str()))?;
 
             // Prevent stdin / stdout from being closed
             mem::forget(stdin);
             mem::forget(stdout);
         }
+        Command::Walltime {
+            iterations_per_scenario,
+        } => {
+            let mut timings = vec![Vec::with_capacity(iterations_per_scenario); benchmarks.len()];
+            for _ in 0..iterations_per_scenario {
+                for (i, bench) in benchmarks.iter().enumerate() {
+                    let start = Instant::now();
+
+                    let resumption_kind = black_box(bench.kind.resumption_kind());
+                    let params = black_box(&bench.params);
+
+                    let (mut client_writer, mut server_reader) = async_io::async_pipe(262144);
+                    let (mut server_writer, mut client_reader) = async_io::async_pipe(262144);
+
+                    let server_side = async move {
+                        let handshake_buf = &mut [0u8; 262144];
+                        run_bench(
+                            ServerSideStepper {
+                                io: StepperIo {
+                                    reader: &mut server_reader,
+                                    writer: &mut server_writer,
+                                    handshake_buf,
+                                },
+                                config: ServerSideStepper::make_config(params, resumption_kind),
+                            },
+                            bench.kind,
+                        )
+                        .await
+                    };
+
+                    let client_side = async move {
+                        let handshake_buf = &mut [0u8; 262144];
+                        run_bench(
+                            ClientSideStepper {
+                                io: StepperIo {
+                                    reader: &mut client_reader,
+                                    writer: &mut client_writer,
+                                    handshake_buf,
+                                },
+                                resumption_kind,
+                                config: ClientSideStepper::make_config(params, resumption_kind),
+                            },
+                            bench.kind,
+                        )
+                        .await
+                    };
+
+                    async_io::block_on_concurrent(client_side, server_side)
+                        .with_context(|| format!("{} crashed", bench.name()))?;
+
+                    let duration = Instant::now() - start;
+                    timings[i].push(duration);
```

Comment:
I think this can be simplified with `elapsed()`:
```suggestion
                    timings[i].push(start.elapsed());
```

Reply:
TIL, thanks!